src/main/java/me/tsukanov/counter/ui/dialogs/AddDialog.java

```diff
@@ -43,23 +43,33 @@ public Dialog onCreateDialog(Bundle savedInstanceState) {
                         new DialogInterface.OnClickListener() {
                             public void onClick(DialogInterface dialog, int which) {
                                 String name = nameInput.getText().toString();
-                                if (name.equals("")) {
-                                    Toast.makeText(activity,
-                                            getResources().getText(R.string.toast_no_name_message),
-                                            Toast.LENGTH_SHORT).show();
+                                name = name.trim();
+                                CounterApplication app = (CounterApplication) activity.getApplication();
+                                if (name.equals("") || name.length() == 0) {
+
+                                    name ="New counter";
+
+                                }
+
+                                int i = 2;
+                                String tempName = name;
+                                while(app.counters.containsKey(tempName)){
+                                    tempName =name + "_" + Integer.toString(i);
```

Comment:
Can you clarify how this works? I'm not sure if I understand if implementation is correct.

It might help to extract counting of duplicates into a separate function.

Reply:
Well, we search if the counter contains the given name and if it does we rename it as shown. But then, we have to search if the new name we gave is also contained in counter. If we have "counter" and "counter_2" and we try to enter "counter" we want it to be renamed to "counter_3" not "counter_2" and thats why we keep looping.